src/overseer/heartbeat.clj

```diff
@@ -65,13 +65,21 @@
 (defn- reset-job-txns
   "Given a job entity that has failed heartbeat, return txns to
    reset its status to unstarted and retract its heartbeat (to prevent
-   erroneous reset by another monitor)
-
-   NOTE: cas can throw, monitor responsible for handling properly"
+   erroneous reset by another monitor)"
   [{ent-id :db/id heartbeat :job/heartbeat :as job}]
   [[:db/retract ent-id :job/heartbeat heartbeat]
    [:db.fn/cas ent-id :job/status :started :unstarted]])
 
+(defn cas-failed?
+  "Return boolean whether an exception is specifically a Datomic
+  check-and-set failed, :db.error/cas-failed"
```

Comment:
nit I think the type and value are sufficiently described in the name/impl, `Return whether an exception is a Datomic check-and-set failure`